client/src/components/markdown/RenkuMarkdownWithPathTranslation.tsx

```diff
@@ -28,30 +29,72 @@ const REF_TYPES = {
   FILE_PREV: "file", // with ! in the front
 };
 
-const getFileExtension = (file_path) => {
-  if (!file_path) return null;
+type IFileRef = {
```

Comment:
Naming is not conveying what the type is.

```suggestion
type FileReference = {
```